### PR TITLE
Refactor and Reorganize code in plugin

### PR DESCRIPTION
**Pull request by reachjainrahul**

### CODE REVIEW

File: pkg/cloudprovider/securitygroup/securitygroup.go [hunk 1]
```diff
@@ -101,194 +89,205 @@ Antrea internal NetworkPolicy To SecurityGroup Mapping strategy
...
+	// 1. Controller calls it at initialization to obtains the channel.
+	// 2. Controller waits on channel returned in 1, and expects that when channel wakes up it return the entire SGs configured.
+	// 3. Plug-in shall wake up the channel initially after sync up with the cloud; and then periodically.
+	// 4. Controller, upon receive entire SGs set, proceed to reconcile between K8s configuration and cloud configuration.
+	// This API ensures cloud plug-in stays stateless.
+	// - Correct SGs accidentally changed by customers via cloud API/console directly.
+	GetSecurityGroupSyncChan() <-chan cloudresource.SynchronizationContent
+}
+
+type SecurityGroupImpl struct{}
```
Comment (Anandkumar26) [inline pkg/cloudprovider/securitygroup/securitygroup.go:136]: This struct can be renamed to CloudSecurityGroupImpl as it is implementing CloudSecurityGroupInterface.
Can be done as part of subsequent cleanup.

Another type with same name(securityGroupImpl) is defined in networkpolicy.go

**Review by Anandkumar26 — COMMENTED** [review]
Not totally convinced with package alignment for pkg/cloudprovider/cloud.
I feel like pkg/cloudprovider/cloud/cloud.go should reside along aws/azure/internal.


And the pkg/cloudprovider/cloudapi can be renamed to pkg/cloudprovider/cloud.

**Review by Anandkumar26 — DISMISSED** [review]
/LGTM

File: pkg/cloudprovider/cloud/cloud.go [hunk 2]
```diff
@@ -78,21 +71,76 @@ type ComputeInterface interface {
...
+// All registered crdv1alpha1 providers.
+var (
+	providersMutex   sync.Mutex
+	providers        = make(map[runtimev1alpha1.CloudProvider]CloudInterface)
+	corePluginLogger = func() logging.Logger {
+		return logging.GetLogger("core-plugin")
+	}
+)
+
+// Register aws cloud.
```
Comment (archanapholla) [inline pkg/cloudprovider/cloud/cloud.go:102]: Comment should include azure too?

File: pkg/cloudprovider/securitygroup/securitygroup.go (hunk 1 above)
Comment (reachjainrahul) [reply pkg/cloudprovider/securitygroup/securitygroup.go]: Done

File: pkg/cloudprovider/cloud/cloud.go (hunk 2 above)
Comment (reachjainrahul) [reply pkg/cloudprovider/cloud/cloud.go]: Done

**Review by archanapholla — APPROVED** [review]
/LGTM

**Pull request merged**